### DISCUSSION TITLE
How to update DepthMaterial of Text component

### DISCUSSION THREAD

**Discussion by Endrickmusic** [pmndrs/drei#1806] [Q&A]
Hi!

I wrote a custom shader for the [Text](https://github.com/pmndrs/drei?tab=readme-ov-file#text) component and tried to update the [MeshDepthMaterial ](https://threejs.org/docs/#api/en/materials/MeshDepthMaterial) as well, to get the right shadows, but with no success. I've set it up in this [Sandbox](https://codesandbox.io/p/github/Endrickmusic/drei-text-shadow-minimal/main?file=%2F.codesandbox%2Ftasks.json&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cls0hh8da00062a654cfd1ybx%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cls0hh8da00022a65vg31qgj6%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cls0hh8da00042a65mvdrc4z3%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cls0hh8da00052a65wefeqpa7%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cls0hh8da00022a65vg31qgj6%2522%253A%257B%2522id%2522%253A%2522cls0hh8da00022a65vg31qgj6%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls0hh8da00012a65l6mgktcw%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cls0hh8da00012a65l6mgktcw%2522%257D%252C%2522cls0hh8da00052a65wefeqpa7%2522%253A%257B%2522id%2522%253A%2522cls0hh8da00052a65wefeqpa7%2522%252C%2522activeTabId%2522%253A%2522cls0lqtpq00d92a657bfr9rnk%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522cls0hkxrm00hb2a65vjovno1u%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522cls0lqtpq00d92a657bfr9rnk%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522cls0hh8da00042a65mvdrc4z3%2522%253A%257B%2522id%2522%253A%2522cls0hh8da00042a65mvdrc4z3%2522%252C%2522activeTabId%2522%253A%2522cls0hh8da00032a65iq4bkrfv%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls0hh8da00032a65iq4bkrfv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522cls0lqgra000ddqf58do261zf%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522CSB_RUN_OUTSIDE_CONTAINER%253D1%2520devcontainer%2520templates%2520apply%2520--template-id%2520%255C%2522ghcr.io%252Fdevcontainers%252Ftemplates%252Fjavascript-node%255C%2522%2520--template-args%2520%27%257B%257D%27%2520--features%2520%27%255B%255D%27%2522%252C%2522id%2522%253A%2522cls0hicj800502a65wsx0345c%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522cls0hidha00722a65m5d9dhn6%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) How can I address the MeshDepthMaterial of Text?

**Comment by faustogenga** [comment]
Try using other Meshes, I noticed that some propetries work with some meshes and not others. For example the color={"#something"} only works with meshBasicMaterial and not meshStandardMaterial, when both meshes have the same available property according to the following documents : https://threejs.org/docs/?q=mes#api/en/materials/MeshBasicMaterial. Play around with these docs to see what works and what doesnt.
